queue.py

```diff
@@ -24,7 +24,8 @@ def enqueue(self, val):
             self.head = Node(val)
```

Comment:
Using 0 as a "falsey" value can be dangerous - It has a lot of little liabilities that can come up as your refactor, since this would evaluate to true if size was an empty string or None a number of other things. In this particular funciton it is probably fine, but in the future keep in mind the issues that could arise with falsey values.